source/agora/utils/Test.d

```diff
@@ -624,62 +627,274 @@ unittest
+
+        // Get the hash to sign
+        const txHash = this.data.hashFull();
+        // Sign all inputs using WK keys
+        foreach (idx, ref in_; this.inputs)
+        {
+            auto ownerKP = getOwnerKeyPair(in_.output.address, unknowns);
+            assert(ownerKP !is KeyPair.init,
+                    "Address not found in Well-Known keypairs: "
+                    ~ in_.output.address.toString());
+            auto signature = () @trusted { return ownerKP.secret.sign(txHash[]); }();
+            this.data.inputs[idx].signature = signature;
+        }
+
+        // Return the result and reset this
+        this.inputs = null;
+        scope (success) this.data = Transaction.init;
+        return this.data;
+    }
+
+    /***************************************************************************
 
-    auto remainder = amount.div(toward.length);
-    Transaction result = Transaction(type);
-    foreach (addr; toward)
-        result.outputs ~= Output(amount, addr);
+        Finds a well-known `KeyPair` corresponding to the `PublicKey`.
 
-    // Add the remainder to the first output.
-    result.outputs[0].value.mustAdd(remainder);
-    const inputHash = input.hashFull();
+        Params:
+            address = finding `PublicKey`
+            unknowns =  unknown `KeyPair`s
 
-    // Add support for Transactions with multiple recipients
-    foreach (idx, const ref _; input.outputs)
-        result.inputs ~= Input(inputHash, cast(uint) idx);
+        Returns:
+            returns KeyPair.init if `address` is not found.
 
-    scope sign = (KeyPair kp, Hash h) @trusted { return kp.secret.sign(h[]); };
+    ***************************************************************************/
 
-    const resultHash = result.hashFull();
-    foreach (idx, ref in_; result.inputs)
+    private KeyPair getOwnerKeyPair (scope const PublicKey address,
+        scope const(KeyPair)[] unknowns = null) @safe
     {
-        auto rng = from.find!(a => a.address == input.outputs[idx].address);
-        assert(rng.length);
-        const owner = rng[0];
-        in_.signature = sign(owner, resultHash);
+        version(unittest)
+        {
+            auto gen_keypair = () @trusted {
+                return getGenesisKeyPair();
+            }();
+
+            if (address == gen_keypair.address)
+                return gen_keypair;
+        }
+        if (unknowns.length > 0)
+        {
+            auto index = unknowns.find!(res => res.address == address);
+            if (!index.empty)
+                return index.front;
+        }
+        return WK.Keys[address];
+    }
+
+    /***************************************************************************
+
+        Splits the attached input into multiple outputs.
+
+        Params:
+            toward = address of destination to be transferred.
+
+    ***************************************************************************/
+
+    public void split (scope const PublicKey[] toward...) @safe
```

Comment:
```suggestion
    public ref typeof(this) split (scope const PublicKey[] toward...) @safe return
```

This will allow us to do `auto tx = TxBuilder(...).split(...).merge(...).split(...).sign();`